src/WebJobs.Script.WebHost/Metrics/LinuxContainerLegionMetricsPublisher.cs

```diff
@@ -75,21 +76,7 @@ public LinuxContainerLegionMetricsPublisher(IEnvironment environment, IOptionsMo
             _hostingConfigOptionsOnChangeListener = _hostingConfigOptions.OnChange(OnHostingConfigOptionsChanged);
         }
 
-        public IMetricsLogger MetricsLogger
-        {
-            get
-            {
-                if (_metricsLogger == null)
-                {
-                    if (!Utility.TryGetHostService<IMetricsLogger>(_scriptHostManager, out _metricsLogger))
-                    {
-                        throw new InvalidOperationException($"Unable to resolve {nameof(IMetricsLogger)} service.");
-                    }
-                }
-
-                return _metricsLogger;
-            }
-        }
+        private IMetricsLogger MetricsLogger => _metricsLogger ??= _serviceProvider.GetRequiredService<IMetricsLogger>();
```

Comment:
`GetRequiredService` throws an `InvalidOperationException` when the service is not available in container.

Reply:
The message will change though. I am not sure, but do we have any detector/alert that will rely on this message?

Reply:
I do not think there are any detectors as of today. This class was added in #10932 and it's not shipped yet.